calibration of frequency and samplerate(s) ..

hayguen opened the pull request:
.. for next hdsdr release (work n progress)

* calibration setting (in registry of hdsdr) modifies reported samplerates
 see https://github.com/ik1xpv/ExtIO_sddc/issues/110
 calibration settings has to be entered manually in the registry
 Computer\HKEY_CURRENT_USER\Software\HDSDR\ExtIO_SDDC.dll\012_Value
* calibration of hdsdr can be used to determine the ppm value ..
 but it must be resetted, when entering this value in the registry
 - whilst hdsdr is already terminated
* added new extio function 'ExtIoSrateSelText()'
 to display a readable samplerate in next hdsdr release
* added double variants for several extio functions
* deactivated unnecessary extio Tune functions

Signed-off-by: hayati ayguen <h_ayguen@web.de>

ik1xpv commented:
I tested it with old HDSDR 2.80 and new beta. It's nice:
![image](https://user-images.githubusercontent.com/9883800/103918672-5f4a5b80-510f-11eb-84b8-aaa448aba6c6.png)
Can we marge it ?

howard0su commented on Core/fft_mt_r2iq.cpp:
```diff
@@ -46,16 +47,21 @@ r2iqControlClass::r2iqControlClass()
+	for (unsigned k = 0; k < 65536; k++)
 	{
-		if ((k & 1) == 1)
+		if ((k & 1))
```
Remove a pair of ()

howard0su commented on Core/fft_mt_r2iq.cpp:
```diff
@@ -188,10 +194,20 @@ void fft_mt_r2iq::Init(float gain, int16_t **buffers, float** obuffers)
 		filterplan_t2f_c2c = fftwf_plan_dft_1d(halfFft, pfilterht, filterHw[0], FFTW_FORWARD, FFTW_MEASURE);
 		float *pht = new float[halfFft / 4 + 1];
-		for (int d = 0; d < NDECIDX; d++)
+		for (int d = 0; d < NDECIDX; d++)	// @todo when increasing NDECIDX
```
what does it mean about todo? I have another PR to increase NDECIDX

hayguen replied:
> what does it mean about todo? I have another PR to increase NDECIDX

just wanted to note, that here (in the switch below) there's some work necessary (todo), when increasing NDECIDX. wanted to have something visible when looking for all occurences of NDECIDX.

howard0su replied:
i see. i have the change. will submit soon.

howard0su commented on Core/fft_mt_r2iq.cpp:
```diff
@@ -188,10 +194,20 @@ void fft_mt_r2iq::Init(float gain, int16_t **buffers, float** obuffers)
 		{
+			// @todo: have dynamic bandpass filter size - depending on decimation
+			//   to allow same stopband-attenuation for all decimations
+#if 0
```
please remove deadcode.

howard0su commented on Core/fft_mt_r2iq.cpp:
```diff
@@ -302,6 +325,9 @@ void * fft_mt_r2iq::r2iqThreadf(r2iqThreadArg *th) {
 		{
+			// @todo: can this get implemented without the RandTable[]?
+			//   for less cache trashing?
+			//   ideally with some simd commands ..
```
agree

howard0su reviewed (commented):
I suggest you split this PR into two PRs. One for extio, another for the comments and todo you added for r2iq.

hayguen commented:
> I suggest you split this PR into two PRs

oops. pushing the second commit onto my github .. github automatically put it into the previous pull request - which wasn't my intent

hayguen commented:
moved the second commit into the branch https://github.com/hayguen/ExtIO_sddc/tree/r2c_comments
and removed that commit from my master.
now, this pull request shouldn't contain the commit with all the comments .. and i could create a seperate pull request for that one
